### PR TITLE
Match all tokens - Ignore position between them

### PR DESCRIPTION
**Pull request by joaoalves-kununu**

### CODE REVIEW

File: angucomplete-alt.js [hunk 1]
```diff
@@ -217,20 +217,43 @@
...
-        matches = target.match(re);
-        if (matches) {
-          result = target.replace(re,
-              '<span class="'+ scope.matchClass +'">'+ matches[0] +'</span>');
+
+        if (scope.matchAllTokens) {
+
+          // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
+          // Escape user input to be treated as a literal string within a regular expression
+          re = new RegExp('('+str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&').match(/\S+/g).join('|')+')', 'gi');
```
Comment (FreaKzero) [inline angucomplete-alt.js:233]: It would be nice to explain in a codecomment what happens here - since its hard to follow due to string concatenation, replace, matching and joining

**Review by FreaKzero — APPROVED** [review]
Besides my minor comment:
Code LGTM :+1:

**Comment by joaoalves-kununu** [comment]
@FreaKzero 

I added a comment :)
